lib/backend/k8s/conversion/conversion.go

```diff
@@ -313,14 +300,16 @@ func (c Converter) K8sNetworkPolicyToCalico(np *extensions.NetworkPolicy) (*mode
 
 // k8sSelectorToCalico takes a namespaced k8s label selector and returns the Calico
 // equivalent.
-func (c Converter) k8sSelectorToCalico(s *metav1.LabelSelector, ns *string) string {
-	// If this is a podSelector, it needs to be namespaced, and it
-	// uses a different prefix.  Otherwise, treat this as a NamespaceSelector.
-	selectors := []string{}
-	prefix := "pcns."
-	if ns != nil {
-		prefix = ""
-		selectors = append(selectors, fmt.Sprintf("calico/k8s_ns == '%s'", *ns))
+func (c Converter) k8sSelectorToCalico(s *metav1.LabelSelector, selectorType selectorType) string {
+	// If this is a namespace selector, the labels need to be prefixed with the profile namespace
+	// prefix.
+	var selectors []string
+	var prefix string
+	if selectorType == SelectorNamespace {
+		prefix = NamespaceLabelPrefix
+	}
+	if selectorType == SelectorPod {
```

Comment:
This isn't quite right - the pod selector needs to be ANDed with that pod's namespace as well, not just with the entire cluster.

We still need this selector: `("projectcalico.org/namespace == '%s', *ns)`

Reply:
Also, I think _all_ selectors generated by this function need the `projectcalico.org/orchestrator == 'k8s'` bit.

Reply:
Ah but policy now has a namespace, so this rule does not need to be part of the policy selector.

However, instead - in the conversion to the v1 model - that's where we insert the additional selector to match on the namespace.

Reply:
(at least that's my thinking)

Reply:
Ok, so looks like the `namespace == foo` bit is added in the networkpolicy processor, is that right?

That makes sense.  I still think all of them need the orchestrator label

Reply:
yup - i think you are correct about the orchestrator -- adding now